core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -148,6 +148,39 @@ public void testScriptsWithoutExtensions() throws IOException {
         }
     }
 
+    public void testHiddenFileSkipped() throws IOException {
+        ContextAndHeaderHolder contextAndHeaders = new ContextAndHeaderHolder();
+        buildScriptService(Settings.EMPTY);
+
+        logger.info("--> setup one hidden test file");
+        Path testFileHidden = scriptsFilePath.resolve(".hidden_file");
+        Path testRegularFile = scriptsFilePath.resolve("test_file.tst");
+        Streams.copy("test_hidden_file".getBytes("UTF-8"), Files.newOutputStream(testFileHidden));
+        Streams.copy("test_file".getBytes("UTF-8"), Files.newOutputStream(testRegularFile));
+        resourceWatcherService.notifyNow();
+
+        try {
+            logger.info("--> verify if hidden_file was skipped");
+            scriptService.compile(new Script("hidden_file", ScriptType.FILE, "test", null),
```

Comment:
Alright! Yep, I think I can find a way out from this point.
I let you know. Thanks a lot!